apps/site/assets/ts/schedule/components/schedule-finder/ServiceSelector.tsx

```diff
@@ -111,23 +162,43 @@ export const ServiceSelector = ({
   routePatterns
 }: Props): ReactElement<HTMLElement> | null => {
   const [selectedServiceId, setSelectedServiceId] = useState("");
-  const [isLoading, setIsLoading] = useState(true);
-  const [selectedServiceSchedule, setSelectedServiceSchedule] = useState(null);
+  const [state, dispatch] = useReducer(reducer, {
+    data: null,
+    isLoading: true,
+    error: false
+  });
 
   useEffect(
-    () =>
-      fetchSchedule(
-        services,
-        selectedServiceId,
-        routeId,
-        stopId,
-        directionId,
-        setIsLoading,
-        setSelectedServiceSchedule
-      ),
-    [services, directionId, routeId, selectedServiceId, stopId]
+    () => {
+      const selectedService = services.find(
+        service => service.id === selectedServiceId
+      );
+
+      if (!selectedService) {
+        return;
+      }
+      fetchData(routeId, stopId, selectedService, directionId, dispatch);
+    },
+    [services, routeId, directionId, stopId, selectedServiceId]
   );
 
+  //   const [isLoading, setIsLoading] = useState(true);
```

Comment:
@meagonqz this block of commented-out stuff can go

Reply:
Yep I already removed this in a downstream commit from this one 👍